sdk/identity/azure-identity/src/main/java/com/azure/identity/AuthenticationRequiredException.java

```diff
@@ -18,7 +18,7 @@
      * @param message The exception message.
      * @param request The details of the authentication request.
      */
-    public AuthenticationRequiredException(String message, TokenRequestContext request) {
+    AuthenticationRequiredException(String message, TokenRequestContext request) {
```

Comment:
Should you leave the members public since the class is internal. Less to update later to make it public again.

Reply:
This will be a checkstyle  (code style rule) violation  and will fail the CI, if I do that.
Will need to suppress this rule for this class to leave it as public.
I can make a //TODO note here to enable in future., that will be less work.

Reply:
Added TODOs.

Reply:
An alternative is moving the class to implementation package which you are doing for AuthenticationRecord class?